core/src/main/java/org/bitcoinj/core/LegacyAddress.java

```diff
@@ -91,7 +94,30 @@ public static LegacyAddress fromPubKeyHash(NetworkParameters params, byte[] hash
      * @return constructed address
      */
     public static LegacyAddress fromKey(NetworkParameters params, ECKey key) {
-        return fromPubKeyHash(params, key.getPubKeyHash());
+        return fromKey(params, key, ScriptType.P2PKH);
+    }
+
+    /**
+     * Construct a {@link LegacyAddress} that represents the public part of the given {@link ECKey}. Note that an address is
+     * derived from a hash of the public key and is not the public key itself.
+     *
+     * @param params
+     *            network this address is valid for
+     * @param key
+     *            only the public part is used
+     * @param scriptType
+     *            script type. P2PKH and P2SH supported.
+     * @return constructed address
+     */
+    public static LegacyAddress fromKey(NetworkParameters params, ECKey key, Script.ScriptType scriptType) {
+        if (ScriptType.P2PKH == scriptType) {
+            return fromPubKeyHash(params, key.getPubKeyHash());
+        } else if(ScriptType.P2SH == scriptType) {
```

Comment:
From the context you can deduce that this is P2SH-P2WPKH, but maybe this should be in the comments.

I'm not sure if creating an explicit `Script.ScriptType` for `P2SH_P2WPKH` would be a good idea, but thought I'd at least throw the idea out. @schildbach ?